client/src/game/ui/menu/menu.vue

```diff
@@ -223,8 +223,9 @@ export default class MenuBar extends Vue {
     }
 
     nameMarker(marker: string): string {
-        if (layerManager.UUIDMap.get(marker) !== undefined) {
-            return layerManager.UUIDMap.get(marker)!.name;
+        const mrk = layerManager.UUIDMap.get(marker)!.name;
+        if (mrk !== undefined) {
+            return mrk;
```

Comment:
```suggestion
        const shape = layerManager.UUIDMap.get(marker);
        if (shape !== undefined) {
            return shape.name;
```
I don't know why you didn't just use the suggestion I proposed. Your current approach will still error on that first line if the returned value is undefined.

Reply:
This was the only way I could get it working..... Your suggestion gives me an error: 
> Function lacks ending return statement and return type does not include 'unde
> fined'.

Reply:
It's only the start that needs to be replaced that I posted.
The entire function is

```typescript
nameMarker(marker: string): string {
        const shape = layerManager.UUIDMap.get(marker);
        if (shape !== undefined) {
            return shape.name;
        } else {
            return "";
        }
    }
```

Reply:
Probably my lack of TS-knowledge. I still don't get the difference but hey... it works! ;-)
I'll try to dig up some TS course. It's not really JS... ;-)

Reply:
your first line calls the UUIDMap.get function which can return either a value or undefined.  Then you call .name on that result.  If you do it straight away TS will warn you that this can result in an error if get returns undefined.  You then added a ! to tell TS that it does not need to worry and you're sure you know what you're doing.
Thing is that that is bound to go wrong at one point and thus better avoided.  Here avoiding it is relatively easy by calling .get() and then using an if check to verify whether it is undefined or not.